unitex/src/fr/umlv/unitex/leximir/helper/DelacHelper.java

```diff
@@ -0,0 +1,391 @@
+     */
+    public static ArrayList<String> getDicDelacPath() throws FileNotFoundException, IOException{
+        ArrayList<String> list= new ArrayList<>();
+        File folder = new File(DictionaryPath.allDelac);
+        File[] listOfFiles = folder.listFiles();
+        for (File file : listOfFiles) {
+            if (file.isFile()) {
+                if (file.getName().endsWith("dic")) {
+                    list.add(file.getName());
+                }
+            } 
+        }
+        if(list.isEmpty()){
+            throw new FileNotFoundException("dictionary not found");
+        }
+        return list;
+    }
+    
+    /**
+     * This function get all dictionaries in delac folder and return an Object [][] which contains all information(
+     * POS, lemmaAll, lemma, fSTCode, SynSem, comment, wn_SinSet, lemmaId, dicFile)
+     * @param alldelac if alldelac is true, the function takes all delas in delcs folder, else it takes dictionary selected in configuration
+     * @return List of lemma in Object[][] format
+     * @throws FileNotFoundException
+     * @throws IOException 
+     */
+    public static Object[][] getAllDelacFromDicToObject(boolean alldelac, File dic) throws FileNotFoundException, IOException{
+        List<String> list= new ArrayList<>();
+        if(alldelac){
+            list = getDicDelacPath();
+        }
+        else{
+            list=Arrays.asList(dic.getAbsolutePath().toString());
+        }
+        Delac delac = new Delac();
+        Field[] lf = delac.getClass().getDeclaredFields();
+        int count =0;
+        for(String dela:list){
+            //String path = Utils.getValueXml("pathDelas")+"/"+dela;
+            String path="";
+            if(alldelac)
+            path = DictionaryPath.allDelac+"//"+dela;
+            else
+            path = dela;
+            
+            ArrayList<String> readFile = Utils.readFile(path);
+            for(String s:readFile){
+                if(s.trim().length()>0) {
+                    count++;
+                }
+            }
+            DictionaryPath.dictionary.add(dela);
+        }
+        
+        Object[][] ob = new Object[count][lf.length];
+        int k=0;
+        int lemmaId=0;
+        for(String dela:list){
+            String pOs,lemmaAll,lemma,fSTCode,SynSem,comment,wn_SinSet;
+            
+            String dicFile=dela;
+            String path="";
+            if(alldelac)
+            path = DictionaryPath.allDelac+"//"+dela;
+            else
+            path = dela;
+            
+            ArrayList<String> readFile = Utils.readFile(path);
+            for(String s:readFile){
+                StringBuilder sb = new StringBuilder();
+                for (int i = 1; i < s.length(); i++) {
+                    String str = String.valueOf(s.charAt(i));
+                    if(str.matches("^[a-zA-Z0-9áàâäãåçéèêëíì=îïñóòôöõúùûüýÿæœÁÀÂÄÃÅÇÉÈÊËÍÌÎÏÑÓÒÔÖÕÚÙÛÜÝŸÆŒ._\\s-]+$||[$&+'*,:.;\\[?@#\\]/ |)_(-]")){
```

Comment:
What's the purpose of this regex ?

Reply:
Oh I forgot to delete it I found it in Rojo's code, it is useless (he was tring to prevent the use of any dictionaries that doesn't contain latin letters, but now it's working for all languages)